src/mean.jl

```diff
@@ -29,99 +9,139 @@ julia> srs = SimpleRandomSample(apisrs;popsize=:fpc);
+    combine(gdf, [x, :weights] => ((a, b) -> domain_mean(a, design, b)) => AsTable)
 end
 
 """
-Inner method for `by` for StratifiedSample
 Calculates domain mean and its std error, based example 10.3.3 on pg394 Sarndal (1992)
+
+```jldoctest
+julia> using Survey;
+
+julia> strat = load_data("apistrat");
+
+julia> dstrat = StratifiedSample(strat, :stype; popsize  = :fpc);
+
+julia> mean(:api00, :cname, dstrat) |> first 
+DataFrameRow
+ Row │ cname        mean     SE      
+     │ String15     Float64  Float64 
+─────┼───────────────────────────────
+   1 │ Los Angeles  633.511  21.3912
+```
 """
-function mean(x::AbstractVector, popsize::AbstractVector, sampsize::AbstractVector, sampfraction::AbstractVector, strata::AbstractVector)
-    df = DataFrame(x=x, popsize=popsize, sampsize=sampsize, sampfraction=sampfraction, strata=strata)
-    nsdh = []
-    substrata_domain_totals = []
-    Nh = []
-    nh = []
-    fh = []
-    ȳsdh = []
-    sigma_ȳsh_squares = []
-    grouped_frame = groupby(df, :strata)
-    for each_strata in keys(grouped_frame)
-        nsh = nrow(grouped_frame[each_strata])#, nrow=>:nsdh).nsdh
-        push!(nsdh, nsh)
-        substrata_domain_total = sum(grouped_frame[each_strata].x)
-        ȳdh = Statistics.mean(grouped_frame[each_strata].x)
-        push!(ȳsdh, ȳdh)
-        push!(substrata_domain_totals, substrata_domain_total)
-        popsizes = first(grouped_frame[each_strata].popsize)
-        push!(Nh, popsizes)
-        sampsizes = first(grouped_frame[each_strata].sampsize)
-        push!(nh, sampsizes)
-        sampfrac = first(grouped_frame[each_strata].sampfraction)
-        push!(fh, sampfrac)
-        push!(sigma_ȳsh_squares, sum((grouped_frame[each_strata].x .- ȳdh) .^ 2))
+function mean(x::Symbol, by::Symbol, design::StratifiedSample)
+    function domain_mean(x::AbstractVector, popsize::AbstractVector, sampsize::AbstractVector, sampfraction::AbstractVector, strata::AbstractVector)
+        df = DataFrame(x=x, popsize=popsize, sampsize=sampsize, sampfraction=sampfraction, strata=strata)
+        function calculate_components(x, popsize, sampsize, sampfraction)
+            return DataFrame(nsdh = length(x), nsh = length(x), substrata_domain_totals = sum(x), ȳsdh = mean(x), Nh = first(popsize), nh = first(sampsize),fh = first(sampfraction), sigma_ȳsh_squares = sum((x .- mean(x)).^2))
+        end
+        components = combine(groupby(df, :strata), [:x, :popsize, :sampsize, :sampfraction] => calculate_components => AsTable)
+        domain_mean = sum(components.Nh .* components.substrata_domain_totals ./ components.nh) / sum(components.Nh .* components.nsdh ./ components.nh)
+        pdh = components.nsdh ./ components.nh
+        N̂d = sum(components.Nh .* pdh)
+        domain_var = sum(components.Nh .^ 2 .* (1 .- components.fh) .* (components.sigma_ȳsh_squares .+ (components.nsdh .* (1 .- pdh) .* (components.ȳsdh .- domain_mean) .^ 2)) ./ (components.nh .* (components.nh .- 1))) ./ N̂d .^ 2
```

Comment:
This is probably not great, but I don't know how else to make this line shorter. The formatter does something weird with it, so this is my suggestion:

```suggestion
        domain_var = sum(components.Nh .^ 2 .* (1 .- components.fh) .* (components.sigma_ȳsh_squares
            .+ (components.nsdh .* (1 .- pdh) .* (components.ȳsdh .- domain_mean) .^ 2))
            ./ (components.nh .* (components.nh .- 1))) ./ N̂d .^ 2
```